pkg/controller/persistentvolume/index.go

```diff
@@ -87,15 +89,29 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *api.PersistentVo
 		unboundVolumes := []*api.PersistentVolume{}
 		for _, volume := range volumes {
 			// volume isn't currently bound or pre-bound.
-			if volume.Spec.ClaimRef == nil {
-				unboundVolumes = append(unboundVolumes, volume)
-				continue
-			}
-
 			if isVolumeBoundToClaim(volume, claim) {
```

Comment:
@erinm, no, if `claim.Spec.VolumeName` is set by user, binder won't enter this function at all - it will bind the claim only to the specified PV and it does not need to go through all the other ones.